libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/headlineUiConfigHelper.ts

```diff
@@ -73,3 +95,66 @@ export function getHeadlineUiConfig(
 
     return uiConfig;
 }
+
+export function buildHeadlineVisualizationConfig(
+    insight: IInsightDefinition,
+    visualizationProperties: IVisualizationProperties,
+    settings: ISettings,
+    options: IVisProps,
+): IChartConfig {
+    const { config, customVisualizationConfig } = options;
+
+    const supportedProperties = getHeadlineSupportedProperties(insight, visualizationProperties);
+    const fullConfig = {
+        ...config,
+        ...supportedProperties.controls,
+    };
+
+    return updateConfigWithSettings({ ...fullConfig, ...customVisualizationConfig }, settings);
+}
+
+export function getHeadlineSupportedProperties(
+    insight: IInsightDefinition,
+    visualizationProperties: IVisualizationProperties,
+): IVisualizationProperties<HeadlineControlProperties> {
+    const comparison: IComparison = {
+        ...getDefaultComparisonProperties(insight),
+        ...(visualizationProperties?.controls?.comparison || {}),
+    };
+
+    return {
+        ...visualizationProperties,
+        controls: {
+            ...visualizationProperties.controls,
+            comparison,
+        },
+    };
+}
+
+export function isComparisonDisabled(insight: IInsightDefinition) {
+    const primaryMeasure = insightPrimaryMeasure(insight);
+    const secondaryMeasures = insightSecondaryMeasures(insight);
+
+    return !primaryMeasure || secondaryMeasures?.length !== 1;
```

Comment:
Just a question: Will this condition apply once it will be possible to have more secondary measures than two? Is it ok if there is just one secondary measure? I'm not sure what are the exact conditions when comparison should be disabled.

Reply:
Seems like the isComparisonDisabled method is not good which makes other devs confused and hard to understand. 
In fact, we have only one case with one primaryMeasure and one secondaryMeasure will be enabled the comparison.
Therefore, I think it would be better to change the purpose of the method to `isComparisonEnabled`:

```typescript
export function isComparisonEnabled(insight: IInsightDefinition) {
    const primaryMeasure = insightPrimaryMeasure(insight);
    const secondaryMeasures = insightSecondaryMeasures(insight);

    return primaryMeasure && secondaryMeasures?.length === 1;
}
```

Thank you for your question!